pkg/registries/docker/docker.go

```diff
@@ -66,15 +66,21 @@ type Registry struct {
 
 // NewDockerRegistryWithConfig creates a new instantiation of the docker registry
 // TODO(cgorman) AP-386 - properly put the base docker registry into another pkg
-func NewDockerRegistryWithConfig(cfg *Config, integration *storage.ImageIntegration) (*Registry, error) {
+func NewDockerRegistryWithConfig(cfg *Config, integration *storage.ImageIntegration, transports ...registry.Transport) (*Registry, error) {
 	url := cfg.formatURL()
 	// if the registryServer endpoint contains docker.io then the image will be docker.io/namespace/repo:tag
 	registryServer := urlfmt.GetServerFromURL(url)
 	if strings.Contains(cfg.Endpoint, "docker.io") {
 		registryServer = "docker.io"
 	}
 
-	client, err := registry.NewFromTransport(url, cfg.GetTransport(), registry.Quiet)
+	var transport registry.Transport
+	if len(transports) == 0 {
+		transport = DefaultTransport(cfg)
+	} else {
+		transport = transports[0]
+	}
```

Comment:
```suggestion
	transport := DefaultTransport(cfg)
	if len(transports) > 0 {
		transport = transports[0]
	}
```

Reply:
I didn't do this because then we may call `DefaultTransport(cfg)` unnecessarily, which has some (small) overhead because it creates a new transport and does not just return a pointer.